### DISCUSSION TITLE
`gatsby-plugin-feed` : Which field is unused in query ?

### DISCUSSION THREAD

**Discussion by tenpaMk2** [gatsbyjs/gatsby#37740] [Help] [answered]
Hi, developers. Thanks for your developing.

I want more explanation of the following fields of `query` listed in the [official documentation](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-feed) .

> ```js
> {
>   resolve: `gatsby-plugin-feed`,
>   options: {
>     query: `
>       {
>         site {
>           siteMetadata {
>             title
>             description
>             siteUrl
>             site_url: siteUrl
>           }
>         }
>       }
>     `,
>     feeds: [
>       {
>         serialize: ({ query: { site, allMarkdownRemark } }) => {
>           return allMarkdownRemark.nodes.map(node => {
>             return Object.assign({}, node.frontmatter, {
>               description: node.excerpt,
>               date: node.frontmatter.date,
>               url: site.siteMetadata.siteUrl + node.fields.slug,
>               guid: site.siteMetadata.siteUrl + node.fields.slug,
>               custom_elements: [{ "content:encoded": node.html }],
>             })
>           })
>         },
>
> ...
>
>         output: "/rss.xml",
>         title: "Your Site's RSS Feed",
>         match: "^/blog/",
>         link: "https://feeds.feedburner.com/gatsby/blog",
>       },
>     ],
>   },
> },
> ```

`site.siteMetadata.title` and `site.siteMetadata.description` seem to be unused
( `siteUrl` is used in `serialize` function).
However, when I remove `site.siteMetadata.description` query, output XML is changed.

I'm getting confused.
Can someone please explain which fields is important?

**Comment by gh640** [comment, marked as answer]
As you might know, `gatsby-plugin-feed` uses `rss` npm package for most of the work. I recommend to read the readme of the package and how `gatsby-plugin-feed` actually uses it:

- https://www.npmjs.com/package/rss
- https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-feed/src/gatsby-node.js

**Reply by tenpaMk2** [reply]
@gh640 ありがとう❗

I read the source code of `gatsby-plugin-feed` .
It was difficult for me, but I understood the following

- `title` : Pass through as `feedOptions` for `new RSS()`
- `description` : Pass through as `feedOptions` for `new RSS()` 
- `siteUrl` : Not pass through but used in `serialize()` 
- `site_url` : Pass through as `feedOptions` for `new RSS()` 

Now, I can understand that all above fields are important❗ Thanks❗

**Reply by gh640** [reply]
@tenpaMk2 

Yay, glad to hear that! とても丁寧にありがとうございます。解決されたようでよかったです！
